pkg/worker/worker.go

```diff
@@ -0,0 +1,183 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	"github.com/google/uuid"
+	"goteleport.com/pkg/worker/conf"
+	"goteleport.com/pkg/worker/log"
+)
+
+// Command is a job request with the program name and arguments.
+type Command struct {
+	// Name program path/name
+	Name string
+	// Args program arguments
+	Args []string
+}
+
+// Job represents an arbitrary Linux process schedule by the Worker.
+type Job struct {
+	// ID job identifier
+	ID string
+	// Command pipeline
+	Cmd *exec.Cmd
+	// Status of the process.
+	Status *Status
+}
+
+// Status of the process.
+type Status struct {
+	// Process identifier
+	Pid int
+	// ExitCode of the exited process, or -1 if the process hasn't
+	// exited or was terminated by a signal
+	ExitCode int
+	// Exited reports whether the program has exited
+	Exited bool
+}
+
+// Worker defines the basic operations to manage Jobs.
+type Worker interface {
+	// Start creates a Linux process.
+	//    - command: command to be executed
+	// It returns the job ID and the execution error encountered.
+	Start(command Command) (jobID string, err error)
+	// Stop a running Job which kills a running process.
+	//    - ID: Job identifier
+	// It returns the execution error encountered.
+	Stop(jobID string) (err error)
+	// Query a Job to check the current status.
+	//    - ID: Job identifier
+	// It returns process status and the execution error
+	// encountered.
+	Query(jobID string) (status Status, err error)
+	// Streams the process output.
+	//    - ctx: context to cancel the log stream
+	//    - ID: Job identifier
+	// It returns read chan to stream process stdout/stderr and the
+	// execution error encountered.
+	Stream(ctx context.Context, jobID string) (logchan chan string, err error)
+}
+
+// NewWorker creates a new Worker instance.
+func NewWorker(config conf.Config) Worker {
+	return &worker{
+		logger: log.NewLogger(config),
+		jobs:   make(map[string]*Job),
+	}
+}
+
+// worker implementation.
+type worker struct {
+	// logger is responsible to handle the
+	// stdout and stderr of a running process
+	logger log.Logger
+	// jobs is concurrency safe map to store
+	// the requested jobs
+	jobs map[string]*Job
+	// mtx to control jobs concurret access
+	mtx sync.RWMutex
+}
+
+// Start runs a Linux single command with arguments, pipes are not supported.
```

Comment:
If your command is `sh -c "foo | bar"`, pipes should work
```suggestion
// Start runs a Linux single command with arguments.
```